src/stackedtime/sim_nr.jl

```diff
@@ -141,44 +141,63 @@ function damping_schedule(lambda::AbstractVector{<:Real}; verbose::Bool=false)
 end
 
 # the Armijo rule: C.T.Kelly, Iterative Methods for Linear and Nonlinear Equations, ch.8.1, p.137
-function damping_armijo(; alpha::Real=1e-4, sigma::Real=0.5, lambda_min::Real=0.00001, verbose::Bool=false)
+function damping_armijo(; alpha::Real=1e-4, sigma::Real=0.5, lambda_min::Real=1e-5, lambda_max::Real=1.0, lambda_growth::Real=1.1, verbose::Bool=false)
     α = convert(Float64, alpha)
     σ = convert(Float64, sigma)
     λ_min = convert(Float64, lambda_min)
+    λ_max = convert(Float64, lambda_max)
+    λ_growth = convert(Float64, lambda_growth)
     nF2_it = 0  # iteration number at which nF2 is valid
     nF2 = NaN   # the norm of the residual at the beginning of iteration nF2_it
     return function (it::Int, λ::Float64, nF::Float64, F::AbstractVector{Float64},
         ::Union{Nothing,Factorization,AbstractMatrix{Float64}}=nothing,
         ::Union{Nothing,AbstractVector{Float64}}=nothing
     )
-        # @printf "    it=%d, λ=%g, nF=%g\n" it λ nF
-        it < 1 && return true, 1.0
+        it < 1 && return true, min(1.0, λ_max)  
+
         if nF2_it != it
-            # first time we're called this iteration 
-            nF2 = norm(F, 2)  # store the residual 
+            # First call in this iteration: Store the residual norm
+            nF2 = norm(F, 2)
             nF2_it = it
-            return false, 1.0   # try λ=1.0, a full Newton step, first
+            return false, min(1.0, λ_max)  
         end
+
         if λ < λ_min
-            # λ too small
-            verbose && @warn "Linesearch failed."
+            verbose && @warn "Linesearch failed: λ fell below λ_min."
             return true, λ
         end
+
         if norm(F, 2) < (1.0 - α * λ) * nF2
-            # Armijo test pass => accept the given λ
-            return true, λ
+            # Armijo test passed => accept the given λ
+            new_λ = min(λ * λ_growth, λ_max)  # Gradually increase λ but cap at λ_max
```

Comment:
We should call this solver something else if we are not in fact implementing the armijo rule.